### PR TITLE
[EVTB] New object type RAP Business Event Bindings

### PR DESCRIPTION
**Pull request by huber-nicolas**
We should put an AFF Example Object for EVTB into package TEST_AFF_EXAMPLES

### CODE REVIEW

File: file-formats/evtb/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# EVTB File Format
+This file format is used for the ADT Editor for EVTB
```
Comment (huber-nicolas) [inline file-formats/evtb/README.md:2]: This file format is used by the ADT editor for RAP Business Event Bindings (EVTB).

Comment (huber-nicolas) [inline file-formats/evtb/README.md:2]:
```suggestion
This file format is used by the ADT editor for RAP Business Event Bindings (EVTB).
```
```suggestion
This file format is used for the ADT Editor for EVTB
```

File: file-formats/evtb/type/zif_aff_evtb_v1.intf.abap [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header             TYPE zif_aff_types_v1=>ty_header_60,
+
+      "! <p class="shorttext">Namespace</p>
+      "! Namespace
+      "! $maxLength 15
+      "! $required
+      producer_namespace TYPE c LENGTH 30,
```
Comment (huber-nicolas) [inline file-formats/evtb/type/zif_aff_evtb_v1.intf.abap:51]: why is maxLength 15 but the type is c length 30?

File: file-formats/evtb/evtb-v1.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      "maxLength": 83
+    },
+    "events": {
+      "title": "Events",
+      "type": "array",
+      "items": {
+        "title": "Event",
+        "type": "object",
+        "properties": {
+          "eventVersion": {
```
Comment (huber-nicolas) [inline file-formats/evtb/evtb-v1.json:110]: does it make sense to group the 3 version fields?
versions {
normal,
minor,
patch
}

File: file-formats/evtb/examples/z_aff_example_evtb.evtb.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "formatVersion": "1",
+  "header": {
+    "description": "Test RAP ATC Check",
```
Comment (huber-nicolas) [inline file-formats/evtb/examples/z_aff_example_evtb.evtb.json:4]: is ATC Check correct?

File: file-formats/evtb/type/zif_aff_evtb_v1.intf.abap [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      "! <p class="shorttext">Business Object</p>
+      "! Business object
+      "! $required
+      bo_name            TYPE zif_aff_types_v1=>ty_object_name_30,
+
+      "! <p class="shorttext">Business Object Operation</p>
+      "! Business object operation
+      "! $required
+      "! $maxLength 18
+      bo_operation       TYPE c LENGTH 30,
```
Comment (schneidermic0) [inline file-formats/evtb/type/zif_aff_evtb_v1.intf.abap:74]: Why do you specify another `maxLength` here?

File: file-formats/evtb/examples/z_aff_example_evtb.evtb.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    "originalLanguage": "en"
+  },
+  "producerNamespace": "sap.s4",
+  "producer": "PC59C0E46A5E8B597EB82CB1AA2890FE8",
+  "producerVersion": "0001",
+  "boName": "BusinessUser",
+  "boOperation": "Deleted",
+  "producerTopic": "sap.s4.BusinessUser.Deleted.v*",
```
[the remark this answers is not in the document]
Comment (schneidermic0) [reply file-formats/evtb/examples/z_aff_example_evtb.evtb.json:12]: Talked to object owner, we keep it with the current structure

File: file-formats/evtb/evtb-v1.json (hunk 3 above)
Comment (schneidermic0) [reply file-formats/evtb/evtb-v1.json:110]: All events shall be shown as flat-strucured list. Therefore, we won't create a sub structure.

File: file-formats/evtb/type/zif_aff_evtb_v1.intf.abap [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+      "! <p class="shorttext">Type</p>
+      "! Type
+      producer_topic     TYPE c LENGTH 83,
```
Comment (schneidermic0) [inline file-formats/evtb/type/zif_aff_evtb_v1.intf.abap:78]:
```suggestion
      producer_type     TYPE c LENGTH 83,
```

File: file-formats/evtb/type/zif_aff_evtb_v1.intf.abap [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+      entity_name               TYPE c LENGTH 30,
+
+      "! <p class="shorttext">Entity Event Name</p>
+      "! Entity event name
+      "! $required
+      entity_event_name         TYPE c LENGTH 30,
+
+      "! <p class="shorttext">Registered</p>
+      "! Registered
+      producer_topic_registered TYPE abap_bool,
```
[the remark this answers is not in the document]
Comment (schneidermic0) [reply file-formats/evtb/type/zif_aff_evtb_v1.intf.abap:33]:
```suggestion
      "! <p class="shorttext">Registered</p>
      "! Registered
      producer_type_registered TYPE abap_bool,
```

File: file-formats/evtb/type/zif_aff_evtb_v1.intf.abap [hunk 9]
```diff
@@ -0,0 +1,85 @@
+INTERFACE zif_aff_evtb_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Event</p>
+    BEGIN OF ty_event,
+      "! <p class="shorttext">Event Version</p>
+      "! Event version
+      event_version             TYPE n LENGTH 4,
```
[the remark this answers is not in the document]
Comment (schneidermic0) [reply file-formats/evtb/type/zif_aff_evtb_v1.intf.abap:9]: Field is calculated during first save => not required

**Review by wurzka — APPROVED** [review]
LGTM :)

**Pull request merged**